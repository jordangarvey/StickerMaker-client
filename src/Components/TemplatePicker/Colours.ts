/**
 * Enum defining the colours that a sticker can be
 */
enum Colours {
	Airbnb = "#FD5C63",
	Black = "#000000",
	Deliveroo = "#00C1B2",
	Discord = "#7289DA",
	Dribble = "#EA4C89",
	Facebook = "#3B5998",
	Grey1 = "#333333",
	Instagram = "#E1306C",
	Linkedin = "#0077b5",
	Meetup = "#E0393E",
	Medium = "#00AB6C",
	Messenger = "#0084FF",
	Pinterest = "#BD081C",
	Snapchat = "#FFFC00",
	Twitch = "#6441A5",
	Twitter = "#1DA1F2",
	White = "#FFFFFF",
	Youtube = "#FF0000"
}

export default Colours;
