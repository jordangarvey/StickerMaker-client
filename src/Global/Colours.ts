/** Primary colour */
export const primary = "#18A0FB";
/** Alternate primary colour */
export const primaryAlt = "#E8F5FF"
/** Secondary colour */
export const secondary = "#FFFFFF";
/** Alternate secondary colour */
export const secondaryAlt = "#F5F5F5"
/** Background colour */
export const background = "#F5F5F5";
/** Selection colour */
export const selected = "#333333";

/** Colour for text on primary colours */
export const textPrimary = "#FFFFFF";
/** Colour for text on primary colours */
export const textSecondary= "#333333";
