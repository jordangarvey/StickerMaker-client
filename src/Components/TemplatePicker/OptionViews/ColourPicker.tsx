import React, { FC } from "react";
import { StyleSheet, View } from "react-native";

import Categories from "../Categories";
import Colours from "../Colours";

import Colour from "./Colour";


/**
 * Interface for the Colour Picker props
 */
interface IColourPickerProps {
		/** The currently selected colour */
	currentColour: Colours;
	/** Callback to be fired whenever the text value is changed */
	onPress(category: Categories, option: string): void;
}

/**
 * Component to render the Colour Picker
 */
const ColourPicker: FC<IColourPickerProps> = (props) => (
	<View style={styles.colourPicker}>
		{
			Object.keys(Colours).map((colour: any) => (
				<Colour
					key={Colours[colour]}
					onPress={props.onPress}
					colour={Colours[colour] as Colours}
					selected={Colours[colour] === props.currentColour}
				/>
			))
		}
	</View>
);

const styles = StyleSheet.create({
	colourPicker: {
		height: "100%",
		flexDirection: "column",
		flexWrap: "wrap",
		justifyContent: "space-evenly"
	}
});

export default ColourPicker;
