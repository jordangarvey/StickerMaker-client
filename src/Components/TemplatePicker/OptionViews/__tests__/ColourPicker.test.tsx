import React from "react";

import renderer from "react-test-renderer";

import Colours from "../../Colours";

import ColourPicker from "../ColourPicker";


describe("The Colour Picker componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		function mockFunction() {
			console.log("Test")
		}

		const component = renderer.create(
			<ColourPicker
				currentColour={Colours.Grey1}
				onPress={mockFunction}
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
