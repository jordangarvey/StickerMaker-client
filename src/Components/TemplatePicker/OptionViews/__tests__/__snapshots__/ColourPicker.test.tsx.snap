// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Colour Picker componentâ€™s snapshot renders as expected 1`] = `
Array [
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FD5C63",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "borderColor": "green",
        "borderRadius": 25,
        "borderWidth": 3,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00C1B2",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#7289DA",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#EA4C89",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#3B5998",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#333333",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E1306C",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0077b5",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E0393E",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00AB6C",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0084FF",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#BD081C",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFC00",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#6441A5",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1DA1F2",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FF0000",
        "borderRadius": 25,
        "height": 50,
        "opacity": 1,
        "width": 50,
      }
    }
  />,
]
`;
